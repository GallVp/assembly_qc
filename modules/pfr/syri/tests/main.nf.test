nextflow_process {

    name "Test Process SYRI"
    script "../main.nf"
    config './nextflow.config'
    process "SYRI"

    tag "modules"
    tag "modules_nfcore"
    tag "syri"
    tag "minimap2/align"

    setup {
        run("MINIMAP2_ALIGN") {
            script "modules/pfr/minimap2/align/main.nf"
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                ]
                input[1] = [
                    [id: 'reference'],
                    file(params.test_data['homo_sapiens']['genome']['genome2_fasta'], checkIfExists: true)
                ]
                input[2] = true     // bam_format
                input[3] = false    // cigar_paf_format
                input[4] = false    // cigar_bam
                """
            }
        }
    }

    test("homo_sapiens - genome") {

        when {
            process {
                """
                input[0] = MINIMAP2_ALIGN.out.bam
                input[1] = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                input[2] = file(params.test_data['homo_sapiens']['genome']['genome2_fasta'], checkIfExists: true)
                input[3] = 'B'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - genome - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = MINIMAP2_ALIGN.out.bam
                input[1] = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
                input[2] = file(params.test_data['homo_sapiens']['genome']['genome2_fasta'], checkIfExists: true)
                input[3] = 'B'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
