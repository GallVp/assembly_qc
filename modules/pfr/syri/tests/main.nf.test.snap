{
    "homo_sapiens - genome": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "testsyri.out:md5,fd2a1b676912818960056aa281177bb4"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,05cf55e24c29c2661a2b9301d3c7c2a0"
                ],
                "error": [
                    
                ],
                "syri": [
                    [
                        {
                            "id": "test"
                        },
                        "testsyri.out:md5,fd2a1b676912818960056aa281177bb4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,05cf55e24c29c2661a2b9301d3c7c2a0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-25T21:30:50.653501"
    },
    "homo_sapiens - genome - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "testsyri.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,05cf55e24c29c2661a2b9301d3c7c2a0"
                ],
                "error": [
                    
                ],
                "syri": [
                    [
                        {
                            "id": "test"
                        },
                        "testsyri.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,05cf55e24c29c2661a2b9301d3c7c2a0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-25T21:22:31.028411"
    }
}